tests/clzones_test.cpp

```diff
@@ -1,29 +1,79 @@
 #include <iosfwd>
 #include <vector>
 
+#include "activity_actor_definitions.h"
 #include "cata_catch.h"
 #include "clzones.h"
 #include "item.h"
 #include "item_category.h"
 #include "item_pocket.h"
 #include "map_helpers.h"
+#include "player_helpers.h"
 #include "point.h"
 #include "ret_val.h"
 #include "type_id.h"
 
 static const faction_id faction_your_followers( "your_followers" );
 
+static const itype_id itype_556( "556" );
+static const itype_id itype_ammolink223( "ammolink223" );
+static const itype_id itype_belt223( "belt223" );
+
 static const zone_type_id zone_type_LOOT_DRINK( "LOOT_DRINK" );
 static const zone_type_id zone_type_LOOT_FOOD( "LOOT_FOOD" );
 static const zone_type_id zone_type_LOOT_PDRINK( "LOOT_PDRINK" );
 static const zone_type_id zone_type_LOOT_PFOOD( "LOOT_PFOOD" );
+static const zone_type_id zone_type_zone_unload_all( "zone_unload_all" );
+
+static int count_items_or_charges( const tripoint src, const itype_id &id )
+{
+    int n = 0;
+    map_stack items = get_map().i_at( src );
+    for( const item &it : items ) {
+        if( it.typeId() == id ) {
+            n += it.count();
+        }
+    }
+    return n;
+}
 
 static void create_tile_zone( const std::string &name, const zone_type_id &zone_type, tripoint pos )
 {
     zone_manager &zm = zone_manager::get_manager();
     zm.add( name, zone_type, faction_your_followers, false, true, pos, pos );
 }
 
+TEST_CASE( "zone unloading ammo belts", "[zones][items][ammo_belt][activities][unload]" )
+{
+    avatar &dummy = get_avatar();
+    map &here = get_map();
+
+    clear_avatar();
+    clear_map();
+
+    dummy.setpos( tripoint_east );
+    create_tile_zone( "Unload All", zone_type_zone_unload_all, tripoint_east );
```

Comment:
You need to use absolute coordinates. Something like this:
```suggestion
    tripoint_abs_ms const start = here.getglobal( tripoint_east );
    dummy.set_location( start );
    create_tile_zone( "Unload All", zone_type_zone_unload_all, start.raw() );
```
The other tests work because they don't interact with the reality bubble.

Reply:
Thank you! This fixed the test failure. Is this documented somewhere or do you just happen to have this knowledge from personal experience? And do you have any idea why, even without absolute coordinates, the test was working locally? Either way, thanks again. The test not working consistently between my computer and CI was driving me mad.

Reply:
The zone manager is global so it makes sense for it to use global coordinates. The `tripoint` type is supposed to give you this info, but the zones code (particularly `zone_manager::add()`) hasn't been fully migrated yet.

Using local coordinates can still work if the reality bubble is anchored at zero, so local coord == global coord. That's always the case if you only run one test case in isolation, though I'm not entirely sure why the LTO build with the full test suite exposes this issue more often than the non-LTO one.